server/src/controllers/Events/resolver.ts

```diff
@@ -339,7 +339,7 @@ export class EventResolver {
     const isSubscribedToChapter =
       ctx.user.user_chapters.find(
         (user_chapter) => user_chapter.chapter_id === chapterId,
-      )?.subscribed ?? true;
+      )?.subscribed || ctx.user.auto_subscribe;
```

Comment:
```suggestion
      )?.subscribed ?? ctx.user.auto_subscribe;
```
Otherwise if `subscribed` is `false` then this will fall back to `auto_subscribe`.

Reply:
Generally that was the idea, when `auto_subscribe === true`, it subscribes when joining chapters and RSVPing to event. Regardless of the `chapterUser.subscribed` of the event's chapter.

But it can be changed to give `chapterUser.subscribed` precedence. In such case actually `auto_subscribe` won't be needed here at all, as `chapterUser` is always created before `eventUser`.

Reply:
Ah, right. So, this is a tricky one because it's important to avoid sending emails when a user thinks they've unsubscribed.  But it's also important that we have sensible defaults.

As a user, when I unsubscribe from a Chapter, I expect that Chapter to stop communicating with me.  However, if I then RSVP to an Event, do I want notifications about that specific Event, even though I don't want them from the Chapter generally?  Maybe, but I think we need to let people opt-in.

Another issue with the existing logic is how we communicate what it does.  We'd have to say something like 
> Automatically enable email notifications when you join a chapter or RSVP to an event (even if you are not subscribed to the chapter hosting that event).

vs.

> Automatically enable email notifications from a chapter when you join. 

Since we really want to encourage users to turn on notifications, I think it's important that the setting that does is transparent.

Reply:
I feel like the _RSVP_ notifications and _chapter_ notifications shouldn't impact each other.

If someone turns off _chapter.subscribed_, then they are turning off new event announcements, general announcements, and anything else an organizer might send to everyone.  I don't think that would disable RSVP related email notifications just like it wouldn't un-RSVP a person from an event. 

@ojeytonwilliams 's suggestion to labels clear is probably the only way we can avoid confusion.

Just a first pass at it

Chapter Page
_Would you like to receive email updates for new event and general announcements for this Chapter?_ 

RSVP Page / Form
_Would you like to stop all email updates for this event, including changes and cancellations?_

Reply:
Actually, I updated the suggestions so both are asking about disabling emails, like "turn off" and "stop".  We could very well make them both affirmative, like "enable" and "turn on", but it's probably good to do things in a similar format.

I guess it depends on the form element being used and even the current state of the element.  Like, could we change the language of the label based on the state? For example, if it's on, then ask "Would you like to turn off". If it's off, then ask "Would you like to turn on?".

Since we're talking about "Switches" I wanted to repost the earlier conversation about form element selection and about toggling notifications on and off as it was related to emails.  It looks like we even explored the use of the word "Follow {Chapter Name}" instead of "Subscribe" vs "Unsubscribe". I'm not suggesting we change the schema, just that we might use words for the user that help add to the context.

Facebook uses "Follow" and Meetup uses "Join" and the current UI has "Subscribe" vs "Unsubscribe".

One somewhat confusing thing is when you toggle "Subscribe" on it says "member of the chapter". So, does that mean if you unsubscribe that you're no longer a member? I'd imagine you can be a member of the group, and just not get notifications, but that may open a can of worms for the MVP.  

https://github.com/freeCodeCamp/chapter/issues/107#issuecomment-562374895
https://github.com/freeCodeCamp/chapter/issues/107#issuecomment-562423675
https://mui.com/material-ui/react-switch/#switches-with-formcontrollabel
https://github.com/freeCodeCamp/chapter/issues/107#issuecomment-931996286

Reply:
Also, given the types of custom switch designs where you have "On" and "Off", perhaps using a question format for a label, like I suggested above, isn't good with switches.

I know we're using [Chakra](https://chakra-ui.com/docs/components/switch) not using [Material UI, but their custom switches section](https://mui.com/material-ui/react-switch/#customization) shows some of the less-basic things like having a checkmark or the "On" and "Off" labels.

Reply:
@allella that's a good point about the separation between Chapter and Event notifications.  If we make it clear what turning off Chapter notifications does, then it's probably fine to have clicking on RSVP to tacitly accept that you're going to get emails about that Event.

We will need to make this configurable sooner rather than later, though not necessarily for the MVP.

Also, though I've not tested this properly, certain notifications are sure to overlap with the notifications coming from Google Calendar.  We'll have to experiment a bit to make sure users can get the emails they want.

> I know we're using Chakra, but the Material UI docs show what you can do with custom switches like having a checkmark in the switch, or the "On" and "Off" labels.

We should be careful we don't customize switches too much, but something like what you shared below looks good.  I don't know how straightforward that would be to implement, though.

Finally, I'm not sure which word (follow or subscribe) works best.  That might be something we discover through user testing.

Reply:
So, for this PR, I think the consensus is the following. 

- Chapter subscription should control if a user gets new event notifications
- Events should be subscribed to when a user clicks RSVP, regardless of Chapter subscription

We can handle further notification configuration separately (maybe post MVP).

@gikf @allella does that sound about right?

Reply:
For MVP, I'd agree new event notifications are the minimum. If / when organizers & admin can send an email to all Chapter members for a general announcements, then that would seem to fit under "subscribed to chapter notifications", at least until we break the controls down to smaller bits.

Part of our challenge with the switches for MVP is there may be more than one thing it turns on or off, as with the new events & general announcements example for a _chapter_. 

Eventually, we can have more controls to allow a user / member to be more granular about their notification preferences, as in the screenshots of Facebook and Meetup examples.

For the RSVP, it looks like we don't give the _member_ the option to turn off RSVP-related notifications until after they have hit the RSPV confirmation button.
![image](https://user-images.githubusercontent.com/1777776/193110539-3fa8bbd6-db86-4901-915f-c42f9d130401.png)
That's fine, but eventually it may make sense to have the RSVP form with other options (media waiver, food preferences, and such, and that interface might also make sense to toggle on/off the RSVP notifications.

Reply:
To be honest I'm still not sure what to do with how auto subscribe works for events.

What I'm sure is that currently, when unsubscribing from chapter, user is also unsubscribed from events in that chapter. Because that should be slightly outside of this PR scope, I've created issue - #1639

---

> * Chapter subscription should control if a user gets new event notifications

That's how it is, I believe. User has to be subscribed to chapter to receive emails about newly created events.

> * Events should be subscribed to when a user clicks RSVP, regardless of Chapter subscription

As I understand this, it means always subscribing to events when RSVPing.